yt_dlp/extractor/ixigua.py

```diff
@@ -0,0 +1,118 @@
+            'tag': 'video_car',
+            'like_count': int,
+            'dislike_count': int,
+            'view_count': int,
+            'uploader': '懂车帝原创',
+            'uploader_id': '6480145787',
+            'thumbnail': r're:^https?://.*\.(avif|webp)(?:\?.+)',  # still not sure for regex
+            'timestamp': 1629088414,
+            'duration': 1030,
+        },
+        'skip': 'This Extractor need cookies',
+    }
+
+    def _get_json_data(self, webpage, video_id):
+        js_data = get_element_by_id('SSR_HYDRATED_DATA', webpage)
+        if not js_data:
+            raise ExtractorError('Failed to get SSR_HYDRATED_DATA',)
+
+        return self._parse_json(js_data.replace('window._SSR_HYDRATED_DATA=', ''), video_id, transform_source=js_to_json)
+
+    def _get_media_format(self, media_type, media_json):
+        media_specific_format = {}
+        media_data = []
+        if media_type == "dash_video":
+            media_data = traverse_obj(media_json, ('dynamic_video', 'dynamic_video_list'))
+            media_specific_format = {
+                'format_note': 'DASH',
+                'acodec': 'none',
+            }
+        elif media_type == "dash_audio":
+            media_data = traverse_obj(media_json, ('dynamic_video', 'dynamic_audio_list'))
+            media_specific_format = {
+                'format_note': 'DASH',
+                'vcodec': 'none',
+                'ext': 'mp4a',
+            }
+        elif media_type == "normal":
+            media_data = [traverse_obj(media_json, ('video_list', media)) for media in media_json.get('video_list')]
+
+        return self._get_formats(media_data, media_specific_format)
+
+    def _get_formats(self, media_json, media_specific_format):
+        _single_video_format = [{
+            'url': base64.b64decode(media['main_url']).decode(),
+            'width': int_or_none(media.get('vwidth')),
+            'height': int_or_none(media.get('vheight')),
+            'fps': int_or_none(media.get('fps')),
+            'vcodec': media.get('codec_type'),
+            'format_id': str_or_none(media.get('quality_type')),
+            'filesize': int_or_none(media.get('size')),
+            'ext': 'mp4',
+            **media_specific_format
+        } for media in media_json]
+
+        return _single_video_format
+
+    def _media_selector(self, json_data):
+        formats_ = []
+        for media in json_data:
+            media_data = json_data.get(media)
+            if not isinstance(media_data, dict):
+                continue
+            if media.startswith('dash'):
+                video_format = self._get_media_format('dash_video', media_data)
+                audio_format = self._get_media_format('dash_audio', media_data)
+                formats_.extend(audio_format)
+
+            else:
+                video_format = self._get_media_format('normal', media_data)
+
+            formats_.extend(video_format)
+
+        return formats_
```

Comment:
```py
    def _media_selector(self, json_data):
        for path, override in (
            (('video_list',), {}),
            (('dynamic_video', 'dynamic_video_list'), {'acodec': 'none'}),
            (('dynamic_video', 'dynamic_audio_list'), {'vcodec': 'none', 'ext': 'm4a'}),
        ):
            for media in traverse_obj(json_data, (..., *path, ...)):
                if not media.get('main_url'):
                    continue
                yield {
                    'url': base64.b64decode(media['main_url']).decode(),
                    'width': int_or_none(media.get('vwidth')),
                    'height': int_or_none(media.get('vheight')),
                    'fps': int_or_none(media.get('fps')),
                    'vcodec': media.get('codec_type'),
                    'format_id': str_or_none(media.get('quality_type')),
                    'filesize': int_or_none(media.get('size')),
                    'ext': 'mp4',
                    **override,
                }
```

how did you manage to make this so complicated?

Reply:
Sorry, i am just a newbie here

Reply:
Btw, this code make error when sorted. For workaround, i changed the `yield` to the `return`, i didn't know why it's not worked when sorted if stated as generator